src/vde_backend/retrieveDB.py

```diff
@@ -0,0 +1,221 @@
+	""" 
+	Get power data from Power table in Cassandra
+	> for 1 specific home
+	> specific day
+	"""
+
+	where_clause = "home_id = '{}' and day = '{}'".format(home_id, date)
+	cols = [
+		"home_id", 
+		"day", 
+		"ts", 
+		"p_cons", 
+		"p_prod", 
+		"p_tot"
+	]
+
+	home_df = ptc.selectQuery(
+		cassandra_session, 
+		CASSANDRA_KEYSPACE, 
+		table_name, 
+		cols, 
+		where_clause,
+	)
+
+	return home_df
+
+
+def getLastDate(output_file, home_id):
+	""" 
+	Get the last filename sent to the sftp server in order
+	to know which date to start the new query from.
+	"""
+
+	latest = 0
+	latest_file = None
+	latest_date = None
+
+	for filename in os.listdir(output_file):
+		file_time = os.path.getmtime(os.path.join(output_file, filename))
+		if filename.startswith(home_id) and file_time > latest:
+			latest = file_time
+			latest_file = filename
+
+	if latest_file is not None:
+		print(latest_file)
+		latest_date = pd.Timestamp(latest_file.split("_")[1][:-4])
+
+	return latest_date
+
+
+def getAllHistoryDates(cassandra_session, home_id, table_name, now):
+	""" 
+	For a home, get the first timestamp available in the db, and 
+	from that first date, return the list of dates until now.
+	"""
+
+	# get first date available for this home 
+	where_clause = "home_id = '{}'".format(home_id)
+	cols = ["day"]
+	date_df = ptc.selectQuery(
+		cassandra_session, 
+		CASSANDRA_KEYSPACE, 
+		table_name, 
+		cols, 
+		where_clause, 
+		limit=1,
+		allow_filtering=True,
+		distinct=False
+	)
+
+	all_dates = []
+	if len(date_df) > 0:
+		first_date = pd.Timestamp(date_df.iat[0,0])
+
+		all_dates = getDatesBetween(first_date, now)
+
+	return all_dates
+
+
+def processAllHomes(cassandra_session, config, default_date, now, output_filename):
+	""" 
+	save 1 csv file per home, per day
+	- if no data sent for this home yet, we send the whole history
+	- otherwise, we send data from the last sent date to now
+	"""
+
+	for home_id in config.getIds().keys():
+		latest_date = getLastDate(output_filename, home_id)
+		all_dates = [default_date]
+		if latest_date is None:  # history
+			all_dates = getAllHistoryDates(cassandra_session, home_id, TBL_POWER, now)
+		else:					 # realtime
+			all_dates = getDatesBetween(latest_date, now)
+
+		for date in all_dates:
+			csv_filename = "{}_{}.csv".format(home_id, date)
+			print(csv_filename)
+			home_data = getHomePowerDataFromCassandra(
+				cassandra_session, 
+				home_id, 
+				date, 
+				TBL_POWER
+			)
+			
+			saveDataToCsv(home_data.set_index("home_id"), csv_filename, output_filename)
+
+		print("-----------------------")
+
+
+def processArguments():
+	"""
+	process arguments 
+	argument : sftp config filename
+		-> contains host, port, username, password and destination path
+	"""
+	argparser = argparse.ArgumentParser(
+		description=__doc__,
+		formatter_class=argparse.RawDescriptionHelpFormatter,
+	)
+	argparser.add_argument(
+		"output_filename", 
+		type=str, 
+		help="output filename"
```

Comment:
What is the purpose of this script? If it is to dump CSVs from the database?
If so, the user should be able to specify the time range and the home ID.
You can merge this script with the uncommitted one in `/opt/vde/dump_csv.py` on the backend server.

Reply:
The purpose of this script is to dump csv from the database in many ways : 
- real time (get database history if no data saved yet, or get data between last saved date and now)
- specific day of data
- now also a specific home id and a specific date range

Reply:
I like the idea!
Please, replace `__doc__` with the content of your message in the `ArgumentParser`.